src/Aspire.Hosting/Dcp/DcpExecutor.cs

```diff
@@ -1539,31 +1539,21 @@ static bool HasMultipleReplicas(CustomResource resource)
         }
     }
 
-    private static string NormalizeServiceProducerTargetHost(string targetHost, bool isProxied)
+    private static string NormalizeServiceProducerTargetHost(string targetHost)
     {
-        // When proxied, the individual services are always bound to localhost even if the proxy
-        // is bound to different addresses.
-        if (isProxied)
-        {
-            return "localhost";
-        }
-        else
+        return targetHost switch
         {
-            if (string.IsNullOrEmpty(targetHost) || string.Equals(targetHost, "localhost", StringComparison.OrdinalIgnoreCase))
-            {
-                return "localhost";
-            }
-            else if (IPAddress.TryParse(targetHost, out _))
-            {
-                // Use an IP address as is
-                return targetHost;
-            }
-            else
+            null or "" => "localhost", // Default is localhost
+            var s when string.Equals(s, "localhost", StringComparison.OrdinalIgnoreCase) => "localhost", // Explicitly set to localhost
+            var s when s.Length > 10 && s.EndsWith(".localhost", StringComparison.OrdinalIgnoreCase) => "localhost", // Explicitly set to localhost when using .localhost subdomain
+            var s when IPAddress.TryParse(s, out var ipAddress) => ipAddress switch // The host is an IP address
```

Comment:
nested switch expression, you are now an F# developer

Reply:
I've written lisp (Clojure) before to avoid having to write Java for a hadoop map reduce. I'm a secret functional programming sympathizer.